dlib/console_progress_indicator.h

```diff
@@ -166,21 +174,15 @@ namespace dlib
             ss = out.precision(2);
             out << " (" << cur / target_val * 100. << "%). ";
 
-            if (seconds < 60)
-            {
-                ss = out.precision(0);
-                out << "Time remaining: " << seconds << " seconds.                 \r" << std::flush;
-            }
-            else if (seconds < 60*60)
-            {
-                ss = out.precision(2);
-                out << "Time remaining: " << seconds/60 << " minutes.                 \r" << std::flush;
-            }
-            else 
-            {
-                ss = out.precision(2);
-                out << "Time remaining: " << seconds/60/60 << " hours.                 \r" << std::flush;
-            }
+            const auto hours = std::chrono::duration_cast<std::chrono::hours>(rem_time);
+            const auto minutes = std::chrono::duration_cast<std::chrono::minutes>(rem_time) - hours;
+            const auto seconds = std::chrono::duration_cast<std::chrono::seconds>(rem_time) - hours - minutes;
+            out << "Time remaining: ";
+            if (rem_time >= std::chrono::hours(1))
+                out << hours.count() << "h ";
+            if (rem_time >= std::chrono::minutes(1))
+                out << minutes.count() << "min ";
+            out << seconds.count() << "s.                \r" << std::flush;
```

Comment:
Wouldn't you rather have the single floating point value printed?  Rather than something like "1h 5m 8s"?
E.g. something like this
```
            using hours = duration<double, std::ratio<60 * 60>>;
            using minutes = duration<double, std::ratio<60>>;
            using seconds = duration<double>;
            if (rem_time < minutes(1))
            {
                ss = out.precision(0);
                out << "Time remaining: " << duration_cast<seconds>(rem_time).count() << " seconds.                 \r" << std::flush;
            }
            else if (rem_time < hours(1))
            {
                ss = out.precision(2);
                out << "Time remaining: " << duration_cast<minutes>(rem_time).count() << " minutes.                 \r" << std::flush;
            }
            else 
            {
                ss = out.precision(2);
                out << "Time remaining: " << duration_cast<hours>(rem_time).count() << " hours.                 \r" << std::flush;
            }   
```

Reply:
Honestly, “1h 5min 8s” speaks more to me than “1.09 hours”, that's why I made the change. But if you prefer the floating-point format, I'll revert the change :)

I was also thinking about this format “1 hour 5 minutes 8 seconds”, but it's more verbose and needs singular/plural handling. But again, if you prefer it, I'll do it. Whatever you like the most :)

Reply:
Na that's fine. Not sure what my preference is. Was just not sure you knew about the option to use double with a duration :)

Reply:
Ah, I got it, yes, I was aware of this thing, I did the “h min s” format on purpose :)
I remember watching this a couple of years ago and being completely fascinated by the chrono library:
[CppCon 2016: Howard Hinnant “A ＜chrono＞ Tutorial"](https://www.youtube.com/watch?v=P32hvk8b13M)